src/com/googlecode/jmxtrans/model/output/GraphiteWriter.java

```diff
@@ -101,6 +101,11 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 							log.debug("Graphite Message: {}", line);
 							writer.write(line);
 							writer.flush();
+							if (writer.checkError()) {
+								log.error("Error writing to Graphite, clearing Graphite socket pool");
+								pool.returnObject(address, socket);
+								pool.clear();
```

Comment:
True, but we weren't sure if the other connections were bad as well (for instance, if it was a problem with Graphite dropping the connections), and since this is a production application for us, we felt like taking the hit and reinitializing all the connections would be worth it. The other aspect of this is that it was intermittent, so we'd have to wait for a day or so after deploying a fix to make sure it actually worked, and we needed it fixed ASAP.